db/view/view.cc

```diff
@@ -1301,16 +1304,24 @@ future<> mutate_MV(
                 return make_ready_future<>();
             });
             if (wait_for_all) {
-                fs->push_back(std::move(view_update));
+                remote_update = std::move(view_update);
             } else {
                 // The update is sent to background in order to preserve availability,
                 // its parallelism is limited by view_update_concurrency_semaphore
                 (void)view_update;
             }
         }
+        try {
+            co_await when_all_succeed(std::move(local_view_update), std::move(remote_update));
+        } catch (...) {
+            if (!err) {
+                err = std::current_exception();
+            }
+        }
+    });
+    if (err) {
+        std::rethrow_exception(err);
```

Comment:
I agree that it's sad that after investing so much thinking in the early days of Seastar on having an efficient way to create exceptions *without* throwing, we gave up on this and now we use throw so much.